src/Database/LSMTree/Internal/Run/Index/Compact.hs

```diff
@@ -527,13 +527,20 @@ search :: SerialisedKey -> CompactIndex -> SearchResult
 search k CompactIndex{..} = -- Pre: @[0, V.length ciPrimary)@
     let !rfbits    = fromIntegral $ keyTopBits16 ciRangeFinderPrecision k
         !lb        = fromIntegral $ ciRangeFinder VP.! rfbits
-        !ub        = fromIntegral $ ciRangeFinder VP.! (rfbits + 1)
+        -- the first page we don't need to look at (exclusive upper bound) is
+        -- the one from the next range finder entry.
+        -- If we are in the last entry already, we search all remaining pages,
+        -- so the upper bound is 'maxPage + 1' or simply 'numPages'.
+        !ub | rfbits + 1 < VP.length ciRangeFinder
+            = fromIntegral $ ciRangeFinder VP.! (rfbits + 1)
+            | otherwise
+            = VG.length ciPrimary
```

Comment:
This is equivalent to what was here before, but it means we have to perform this conditional for every search instead of once during construction. Could you explain why you prefer this over the previous?

Reply:
I just saw that I missed that the `+1` is even already described in `format-run.md`.

Both ways should be fine I think, so the change was mostly driven by seeing the `+1` as something unexpected that we should then document.  Given that it is already documented everywhere, that point is irrelevant and I'll drop the commit.